File: SecondYearCS/OOP/CivitasRB/lib/SorpresaPorCasaHotel.rb
require_relative 'sorpresa.rb'
module CivitasRB
  class SorpresaPorCasaHotel < Sorpresa
    attr_accessor :valor
    #Constructores por defecto y con parametro asdfas
    def initialize(valor,texto)
      super(texto)
      @valor = valor
    end

    def aplicarAJugador(act,todos)
      if self.jugadorCorrecto(act, todos)
        self.informe(act,todos)
        arr = todos.at(act).getPropiedades();
        sum = 0
        for x in arr do
          sum += x.numCasas + x.numHoteles
        end
        todos.at(act).modificarSaldo(@valor*sum)
      end
    end

    def to_s
      return super.to_s +  " valor: " + @valor +  " "
    end
    
  end
end

File: SecondYearCS/OOP/Civitas/src/civitas/SorpresaPorCasaHotel.java
package civitas;

import java.util.ArrayList;

public class SorpresaPorCasaHotel extends Sorpresa{
      
    private float valor;
    
    //Resto de sorpresasa
    SorpresaPorCasaHotel(float valor, String texto){
        super(texto);
        this.valor = valor;
    }
    
    /**
     * @brief Se modifica el saldo del jugador con el valor de la sorpresa multiplicado
     * por el número de casas y hoteles del jugador
     */
    @Override
    public void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if( this.jugadorCorrecto(actual, todos) ){
            informe(actual,todos);
            ArrayList<TituloPropiedad> arr = todos.get(actual).getPropiedades();
            int sum = 0;
            for(int i=0;i<arr.size();i++){
                sum += arr.get(i).getNumCasas();
                sum += arr.get(i).getNumHoteles();
            }
            todos.get(actual).modificarSaldo(this.valor * sum);
        }
    }

    @Override
    public String toString(){
       return super.toString() + " valor: " + valor;
    }

}